api/src/main/java/jakarta/config/spi/ConfigBuilder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jakarta.config.spi;
+
+import jakarta.config.Config;
+
+/**
+ * A builder for manually creating a configuration instance.
+ *
+ * @see ConfigProviderResolver#getBuilder()
+ *
+ * @author <a href="mailto:rmannibucau@apache.org">Romain Manni-Bucau</a>
+ * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
+ */
+public interface ConfigBuilder {
+    /**
+     * Add the <a href="ConfigSource.html#default_config_sources"><em>default configuration sources</em></a> to the
+     * configuration being built.
+     *
+     * @return this configuration builder instance
+     */
+    ConfigBuilder addDefaultSources();
+
+    /**
+     * Add all configuration sources which can be <a href="ConfigSource.html#discovery">discovered</a> from this
```

Comment:
I assume `ConfigSource.html#discovery` and `ConfigSource.html#default_config_sources` are yet to be provided. In a separate PR?